it/uk/gov/hmrc/incometaxsubscriptioneligibility/services/ConvertControlListConfigValuesServicesISpec.scala

```diff
@@ -1,55 +1,39 @@
 
 package uk.gov.hmrc.incometaxsubscriptioneligibility.services
 
-import org.scalatest.{Matchers, WordSpecLike}
+import org.scalatestplus.play.PlaySpec
 import play.api.inject.guice.GuiceApplicationBuilder
 import play.api.{Application, Environment, Mode}
+import uk.gov.hmrc.incometaxsubscriptioneligibility.helpers.ControlListConfigTestHelper
 import uk.gov.hmrc.incometaxsubscriptioneligibility.models.controllist.{ControlListParameter, StudentLoans}
 
-class ConvertConfigValuesServicesISpec extends WordSpecLike with Matchers {
-
-  class Setup(config: Map[ControlListParameter, Boolean]) {
-    lazy val app: Application = new GuiceApplicationBuilder()
-      .in(Environment.simple(mode = Mode.Dev))
-      .configure(toConfigList(config))
-      .build
-    lazy val service: ConvertConfigValuesService = app.injector.instanceOf[ConvertConfigValuesService]
-
-    app.stop()
-  }
-
-  def toConfigList(configValues: Map[ControlListParameter, Boolean]): Map[String, String] = {
-    configValues.map {
-      case (param, isEligible) =>
-        s"control-list.${param.configKey}.eligible" -> isEligible.toString
-    }
-  }
-
-  val allControlListParams: Set[ControlListParameter] = ControlListParameter.getParameterMap.values.toSet
-  val testAllTrue: Map[ControlListParameter, Boolean] = allControlListParams.map(
-    param => param -> true
-  ).toMap
-  val testAllFalse: Map[ControlListParameter, Boolean] = testAllTrue.mapValues(_ => false)
+class ConvertControlListConfigValuesServicesISpec extends PlaySpec with ControlListConfigTestHelper  {
 
   "convertConfigValues service" should {
     "return an empty set" when {
       "all config values are true" in new Setup(testAllTrue) {
-        val result = service.convertConfigValues()
-        result shouldBe Set()
+        val result: Set[ControlListParameter] = convertConfigValuesService.convertConfigValues()
+
+        result mustBe Set()
       }
     }
+
     "return a set with a value" when {
       "Student Loans is false" in new Setup(testAllTrue.updated(StudentLoans, false)) {
-        val result = service.convertConfigValues()
-        result shouldBe Set(StudentLoans)
+
+        val result: Set[ControlListParameter] = convertConfigValuesService.convertConfigValues()
+        result mustBe Set(StudentLoans)
       }
     }
+
     "return a set of values with all the control list parameters" when {
       "all config values are false" in new Setup(testAllFalse) {
-        val result = service.convertConfigValues()
-        result shouldBe allControlListParams
+        val result: Set[ControlListParameter] = convertConfigValuesService.convertConfigValues()
+
+        result mustBe allControlListParams
       }
     }
+
     "returns an empty set and ignores the incorrect parameter" when {
```

Comment:
Could this test (and the one below) be refactored to take advantage of the setup class? Doesn't look like anything different is happening here.